### ISSUE TITLE
new loss reimplements 

### ISSUE THREAD

**Issue by mytk2012** [open-mmlab/mmdetection#3854]
**my code:**
import torch

import torch.nn.functional as F
import math
import torch.nn as nn

from ..builder import LOSSES
from .utils import weight_reduce_loss
def my_loss(pred, targets, weight=None,pos_lambda=1, neg_lambda=0.1/math.log(3.5), L=6., loss_weight=4., reduction='mean', avg_factor=None):
    dtype = targets.dtype
 
    device = targets.device
    num_classes = pred.shape[1]
    margin=0.5
    # print(device),cuda:0
    class_range = torch.arange(1, num_classes + 1, dtype=dtype, device=device).unsqueeze(0)
    t = targets.unsqueeze(1)  
    
    pos_ind = (t == class_range)
    
    neg_ind = (t != class_range) * (t >= 0)
   
   
    pos_prob = pred[pos_ind].sigmoid()
    neg_prob = pred[neg_ind].sigmoid()

    neg_q = F.softmax(neg_prob / neg_lambda, dim=0)  # lambda是λ
    neg_dist = torch.sum(neg_q * neg_prob)
  
    if pos_prob.numel() > 0:
        pos_q = F.softmax(-pos_prob / pos_lambda, dim=0)
        pos_dist = torch.sum(pos_q * pos_prob)
        loss= loss_weight * torch.log(
            1. + torch.exp(L * (neg_dist - pos_dist + margin))) / L
    else:
        loss = loss_weight * torch.log(1. + torch.exp(L * (neg_dist - 1. + margin))) / L
    loss_cls = weight_reduce_loss(loss, weight, reduction, avg_factor)
    return loss_cls

@LOSSES.register_module()
class SigmoidDRLoss(nn.Module):
    def __init__(self, use_sigmoid=True, pos_lambda=1, neg_lambda=0.1 / math.log(3.5), L=6.,
                 loss_weight=4.,reduction='mean'):
        super(SigmoidDRLoss, self).__init__()
        assert use_sigmoid is True, 'Only sigmoid focal loss supported now.'
        self.use_sigmoid = use_sigmoid
        self.margin = 0.5
        self.pos_lambda = pos_lambda
        self.neg_lambda = neg_lambda
        self.L = L
        self.reduction = reduction
        self.loss_weight = loss_weight

    def forward(self, pred, targets,weight=None, avg_factor=None,
                reduction_override=None):
    
        assert reduction_override in (None, 'none', 'mean', 'sum')
        reduction = (
            reduction_override if reduction_override else self.reduction)
        loss_cls = self.loss_weight * my_loss(
         pred, targets,weight, pos_lambda=self.pos_lambda, neg_lambda=self.neg_lambda, L=self.L, loss_weight=self.loss_weight, reduction=reduction, avg_factor=avg_factor)
**error**
![image](https://user-images.githubusercontent.com/57787557/94526062-8cad8080-0267-11eb-8927-3671aad9ab70.png)

**Comment by ZwwWayne** [comment]
Hi @mytk2012 ,
What base mode are you using? Could you provide a config for us to reproduce the bug?
Also, we will appreciate it if you could use the implementation questions template.

**Comment by RyanXLi** [comment]
This issue is stale and closed.
